### DISCUSSION TITLE
terminating simulation and saving snapshots

### DISCUSSION THREAD

**Discussion by MingTengCA** [Nek5000/nekRS#524] [Q&A] [answered]
Hey guys,

I have a quick question regarding saving the snapshot in nekRS.
In Nek5000, the simulation can be terminated and the snapshot of the last step would be saved if we use: echo -1 > ioinfo

While running nekRS, I noticed this did not work.
Does anyone know why? or how to teminate simuilation and save snapshot in nekRS?

thank you guys,
Ming

**Comment by stgeke** [comment, marked as answer]
Just create a file `nekrs.upd` containing:

```
checkpoint = true

[GENERAL]
endTime = 0
```

Note, you need to fix https://github.com/Nek5000/nekRS/issues/525 first, otherwise it won't work correctly.

**Reply by MingTengCA** [reply]
thanks stgeke.
This did not work for me.
Once the file (nekrs.upd) created containninng the lines as you recommended in the user folder (run/turbChannel), the running code failed right away without saving the snapshot.

did I missing anything here?

**Comment by stgeke** [comment]
Did you fix the bug I was referring to?

**Reply by MingTengCA** [reply]
Yes, I went to the file you mentioned and commentd that line (Line 422 in nekrs.cpp)

**Reply by MingTengCA** [reply]
I sorted it out.
Once the bug is fixed, nekRS needs to be reinstalled. A snapshot then can be obtained by:
echo 'checkpoint=true' > nekrs.upd
